tensorflow/python/ops/init_ops.py

```diff
@@ -340,3 +341,45 @@ def __call__(self, shape, dtype=dtypes.float32, partition_info=None):
     """Generate a tensor used to initialize a variable."""
     return random_ops._random_walk(shape, self._nonlinearity, dtype,
                                    seed=self._seed)
+
+
+def orthogonal_initializer(gain=1.0, dtype=dtypes.float32, seed=None):
+  """Returns an initializer that generates an orthogonal rotation matrix.
+
+  Initialize a tensor using an orthogonal matrix.
+
+  Args:
+    gain: multiplicative factor to apply to the orthogonal matrix
```

Comment:
On second thoughts, I think keeping the gain is useful because it simplifies the setup of variables (similar to not having to call `reshape`). It's more or less equivalent to the standard deviation for the `random_normal` initializer. Thoughts?